src/components/Feed/Events/FeedEventNftPreviewWrapper.tsx

```diff
@@ -45,16 +45,20 @@ function FeedEventNftPreviewWrapper({ tokenRef, maxWidth, maxHeight }: Props) {
 
   const { showModal } = useModalActions();
 
-  const handleClick = useCallback(() => {
-    showModal({
-      content: (
-        <StyledNftDetailViewPopover>
-          <NftDetailView authenticatedUserOwnsAsset={false} queryRef={token} />
-        </StyledNftDetailViewPopover>
-      ),
-      isFullPage: true,
-    });
-  }, [showModal, token]);
+  const handleClick = useCallback(
+    ({ onClose }: { onClose?: () => void }) => {
+      showModal({
+        content: (
+          <StyledNftDetailViewPopover>
+            <NftDetailView authenticatedUserOwnsAsset={false} queryRef={token} />
+          </StyledNftDetailViewPopover>
+        ),
+        isFullPage: true,
+        onClose,
```

Comment:
I needed to add an `onClose` callback to the modal so I can control the URL location after the user closes the modal.